@here/harp-materials/lib/MapMeshMaterials.ts

```diff
@@ -407,6 +413,183 @@ export class FadingFeatureMixin implements FadingFeature {
     }
 }
 
+export namespace ExtrusionFeature {
+    /**
+     * Patch the THREE.ShaderChunk on first call with some extra shader chunks.
+     */
+    export function patchGlobalShaderChunks() {
+        if (THREE.ShaderChunk.extrusion_pars_vertex === undefined) {
+            Object.assign(THREE.ShaderChunk, extrusionShaderChunk);
+        }
+    }
+
+    /**
+     * Update the internals of the `ExtrusionFeature` depending on the value of [[extrusionRatio]].
+     *
+     * @param ExtrusionMaterial ExtrusionFeature
+     */
+    export function updateExtrusionFeature(extrusionMaterial: ExtrusionFeature): void {
+        extrusionMaterial.needsUpdate = true;
+
+        if (extrusionMaterial.defines === undefined) {
+            extrusionMaterial.defines = {};
+        }
+
+        if (
+            extrusionMaterial.extrusionRatio !== undefined &&
+            extrusionMaterial.extrusionRatio >= AnimatedExtrusionHandler.DEFAULT_RATIO_MAX
+        ) {
+            // Add this define to differentiate it internally from other MeshBasicMaterial
+            extrusionMaterial.defines.EXTRUSION_MATERIAL = "";
+        }
+    }
+
+    /**
+     * This function should be called on implementors of ExtrusionFeature in the `onBeforeCompile`
+     * callback of that material. It adds the required code to the shaders and declares the new
+     * uniforms that control extrusion.
+     *
+     * @param extrusionMaterial Material to add uniforms to.
+     * @param shader [[THREE.WebGLShader]] containing the vertex and fragment shaders to add the
+     *                  special includes to.
+     */
+    export function onBeforeCompile(extrusionMaterial: ExtrusionFeature, shader: any) {
+        if (extrusionMaterial.extrusionRatio === undefined) {
+            return;
+        }
+        // The vertex and fragment shaders have been constructed dynamically. The uniforms and
+        // the shader includes are now appended to them.
+
+        // Create the uniforms for the shader (if not already existing), and add the new uniforms
+        // to it:
+        const uniforms = shader.uniforms as UniformsType;
+        uniforms.extrusionRatio = { value: extrusionMaterial.extrusionRatio };
+
+        shader.vertexShader = insertShaderInclude(
+            shader.vertexShader,
+            "common",
+            "extrusion_pars_vertex"
+        );
+
+        shader.vertexShader = insertShaderInclude(
+            shader.vertexShader,
+            "begin_vertex",
+            "extrusion_vertex",
+            true
+        );
+    }
+
+    /**
+     * @param object [[THREE.Object3D]] to prepare for rendering.
+     * @param additionalCallback If defined, this function will be called before the function will
+     *          return.
+     */
+    export function addRenderHelper(object: THREE.Object3D) {
+        object.onBeforeRender = chainCallbacks(
+            object.onBeforeRender,
+            ExtrusionFeature.onBeforeRender
+        );
+    }
+
```

Comment:
Add doc like this:

Handles animated extrusion on each frame. Should be installed as respective `Object3D.onBeforeRender` of meshes which use animated extusion feature.